pkg/plugin/aggregation/run.go

```diff
@@ -113,21 +116,31 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 	}()
 
 	updater := newUpdater(expectedResults, namespace, client)
-	ticker := time.NewTicker(annotationUpdateFreq)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer func() {
+		logrus.Info("Last update to annotations on exit")
+		// This is the async exit cleanup function.
+		// 1. Stop the annotation updater
+		cancel()
+		// 2. Try one last time to get an update out on exit
+		updater.ReceiveAll(aggr.Results)
+		if err := updater.Annotate(); err != nil {
+			logrus.WithError(err).Info("couldn't annotate sonobuoy pod")
+		}
+	}()
 
 	// 3. Regularly annotate the Aggregator pod with the current run status
-	go func() {
-		defer ticker.Stop()
-		for range ticker.C {
-			updater.ReceiveAll(aggr.Results)
-			if err := updater.Annotate(); err != nil {
-				logrus.WithError(err).Info("couldn't annotate sonobuoy pod")
-			}
-			if aggr.isComplete() {
-				return
-			}
+	wait.JitterUntil(func() {
+		pluginsdone := aggr.isComplete()
+		updater.ReceiveAll(aggr.Results)
+		if err := updater.Annotate(); err != nil {
+			logrus.WithError(err).Info("couldn't annotate sonobuoy pod")
 		}
-	}()
+		if pluginsdone {
```

Comment:
any particular reason we emit this on the JitterUntil case but not on the termination guard case?